### ISSUE TITLE
Kitsas 5.0 kaatuu Qt6Core.dll:ään

### ISSUE THREAD

**Issue by ktdd** [artoh/kitupiikki#1354]
Ongelma sekä Windows 11:ssä että Windows 10 -virtuaalikoneessa. Antaa lisätä pari tositetta jonka jälkeen kaatuu ilman virheilmoitusta, mutta event logissa seuraavat tiedot:

### W10 (VMWare):
Fault bucket 1395316487334481229, type 4
Event Name: APPCRASH
Response: Not available
Cab Id: 0

Problem signature:
P1: kitsas.exe
P2: 0.0.0.0
P3: 6470356f
P4: Qt6Core.dll
P5: 6.5.0.0
P6: 641e963b
P7: c0000005
P8: 000000000031a180
P9: 
P10: 

Analysis symbol: 
Rechecking for solution: 0
Report Id: 24e8d779-3599-4216-a5f5-3d6dd3c7d903
Report Status: 268435456
Hashed bucket: a90a282cbff57fdfc35d28f3d8d14d4d
Cab Guid: 0

### W11:
Faulting application name: kitsas.exe, version: 0.0.0.0, time stamp: 0x6470356f
Faulting module name: Qt6Core.dll, version: 6.5.1.0, time stamp: 0x3e317cce
Exception code: 0xc0000409
Fault offset: 0x0000000000017ee8
Faulting process id: 0x3f60
Faulting application start time: 0x01d9d9ec5263bc67
Faulting application path: A:\Drive\Kitsas5\kitsas.exe
Faulting module path: A:\Drive\Kitsas5\Qt6Core.dll
Report Id: 889f2003-ef61-4a3f-8128-19eaeee8f07e
Faulting package full name: 
Faulting package-relative application ID:

**Comment by ktdd** [comment]
Testasin juuri betaversiota, antaa tallentaa enemmän tositteita mutta lopulta kaatuu kuitenkin.

**Comment by ktdd** [comment]
Kitsas 5.1 sama homma.

Fault bucket 1631675761851004329, type 4
Event Name: APPCRASH
Response: Not available
Cab Id: 0

Problem signature:
P1: kitsas.exe
P2: 0.0.0.0
P3: 64f03930
P4: Qt6Core.dll
P5: 6.5.0.0
P6: 641e963b
P7: c0000005
P8: 000000000031a180
P9: 
P10:

**Comment by helihyv** [comment]
Asensitko ohjelman valmiista paketista vai käänsitkö itse?

**Comment by ktdd** [comment]
Valmis paketti joka kerta. Testasin eilen jopa tuoreinta versiota Ubuntulla, kaatui virheeseen "libQt6Core.so.6 general protection fault".

**Comment by ktdd** [comment]
Edelleen jatkuu. Tässä välissä on tullut formatoituakin eli nyt alkaa olla kaikki keinot käytetty.

**Comment by petriaarnio** [comment]
Oletko kokeillut viimeisimmällä versiolla 5.1.1?

**Comment by ktdd** [comment]
> Oletko kokeillut viimeisimmällä versiolla 5.1.1?

Joo, oon aina päivittänyt julkaisupäivänä siinä toivossa että ei enää kaatuisi. Ei tee yhtään mieli palata takaisin nelosversioon ja liekö edes vitosella tallennettu kirjanpito toimisi.

**Comment by petriaarnio** [comment]
En saa kaatumaan omilla koneilla. Sekä Intel- että M1-versiot.

Missä kohtaa softa kaatuu?

**Comment by petriaarnio** [comment]
Ja nyt tuli sitten tarkemmin luetuksi ettei tämä ollutkaan se Mac-version ketju. Poistun takavasemmalle.

**Comment by ktdd** [comment]
Vastataan nyt siitä huolimatta, että sovellus kaatuu aina samassa vaiheessa - välittömästi tositteen tallentamisen jälkeen. Joskus tositteita voi syöttää ihan huoletta parikymmentä eikä ole ongelmia, eilen kaatuili aina kahden tositteen jälkeen.

**Comment by helihyv** [comment]
Kaatuuko jollain tietyllä tositetyypillä (esim. tiliote, meno/tulo) vai millä tahansa?

**Comment by jrasanen** [comment]
Samankaltaisia virheitä omassa softassa debuganneena c0000005 lienee null-osoittimen dereferointi, ja 000000000031a180 on poikkeuksen heittänyt offset Qt6Core.dll tiedoston alusta. 🤔

**Comment by ktdd** [comment]
Luulin jo että joku Windowsin päivitys tms oli tämän ongelman korjannut, kun syys- ja lokakuun kirjanpitoa tehdessä Kitsas ei kaatunut kertaakaan. Nyt taas marraskuun kirjanpidossa saan tallennettua max kaksi tositetta ennen kaatumista. Kaatuminen tapahtuu joka kerta samassa paikassa, eli tiliotteen syöttö > Tosite > Tietojen täyttö > Valmis > kaatuminen.

**Comment by ktdd** [comment]
Uusinta versiota testattu, kaatui noin kuuden tositteen tallentamisen jälkeen. Seuraavalla käynnistyskerralla kolmen tositteen, sitä seuraavat neljä käynnistyskertaa antoivat tallentaa yhden tositteen per.

**Comment by ktdd** [comment]
Jälleen kirjanpitoa tekemässä uusimmalla sovellusversiolla. Tiliotetta tallentaessa antoi taas tallentaa kymmenkunta tositetta, jonka jälkeen kaatui. Tämän jälkeen antaa tallentaa aina yhden tositteen kerrallaan jonka jälkeen kaatuu.

**Comment by ktdd** [comment]
Myös tuoreimmalla versiolla sama ongelma.
